radix-engine/src/system/transaction/instructions.rs

```diff
@@ -318,6 +323,118 @@ impl TxnNormalInstruction for AssertWorktopContainsNonFungibles {
     }
 }
 
+impl TxnNormalInstruction for AssertBucketContents {
+    fn execute<Y: SystemApi<RuntimeError> + KernelNodeApi + KernelSubstateApi<L>, L: Default>(
+        self,
+        _worktop: &mut Worktop,
+        objects: &mut IntentProcessorObjects,
+        api: &mut Y,
+    ) -> Result<InstructionOutput, RuntimeError> {
+        let bucket = objects.get_bucket(&self.bucket_id)?;
+
+        match self.constraint {
```

Comment:
I think this is interesting but a little different model from what I'm operating on. Since they achieve the same thing can we push this refactor into the next PR when other instructions will try and reuse the same logic.

Reply:
Nevermind, I've just updated it to look closer to your model.